Lib/aifc.py

```diff
@@ -179,23 +181,26 @@ def _read_string(file):
         dummy = file.read(1)
     return data
 
-_HUGE_VAL = 1.79769313486231e+308 # See <limits.h>
-
 def _read_float(f): # 10 bytes
-    expon = _read_short(f) # 2 bytes
+    data = f.read(10)
+    if len(data) != 10:
+        raise EOFError
+    expon = int.from_bytes(data[:2], 'big', signed=True)
     sign = 1
     if expon < 0:
         sign = -1
         expon = expon + 0x8000
-    himant = _read_ulong(f) # 4 bytes
-    lomant = _read_ulong(f) # 4 bytes
-    if expon == himant == lomant == 0:
+    mant = int.from_bytes(data[2:], 'big')
+    if not expon and not mant:
         f = 0.0
     elif expon == 0x7FFF:
-        f = _HUGE_VAL
+        f = sys.float_info.max
```

Comment:
An exponent of `0x7fff` indicates an infinity or NaN. I'd expect to see an exception at this point, but I guess we don't want to change existing behaviour. It might be worth a comment, though.